server/SaveLoad.cpp

```diff
@@ -333,8 +333,12 @@ void LoadGame(const std::string& filename, ServerSaveGameData& server_save_game_
         if (!ifs)
             throw std::runtime_error(UNABLE_TO_OPEN_FILE);
 
-        try {
-            // first attempt binary deserialziation
+        std::string signature(5, '\0');
+        if (!ifs.read(&signature[0], 5))
+            throw std::runtime_error(UNABLE_TO_OPEN_FILE);
+        boost::iostreams::seek(ifs, 0, std::ios_base::beg);
+
+        if (strncmp(signature.c_str(), "<?xml", 5)) {
             freeorion_bin_iarchive ia(ifs);
```

Comment:
would be clearer / helpful to have a comment here like
`// XML file format signature not found; try as binary`

Reply:
Added to all places.